pkg/cmd/stack/translate.go

```diff
@@ -120,131 +117,32 @@ func translateServiceEnvFile(ctx context.Context, svc *model.Service, svcName, f
 }
 
 func translateBuildImages(ctx context.Context, s *model.Stack, options *StackDeployOptions) error {
-	hasBuiltSomething := false
-	for svcName, svcInfo := range s.Services {
-		if svcInfo.Build == nil && len(svcInfo.VolumeMounts) == 0 {
-			continue
-		}
-		buildInfo := svcInfo.Build
-		if !okteto.IsOkteto() && buildInfo.Image == "" {
-			return fmt.Errorf("'build' and 'image' fields of service '%s' cannot be empty", svcName)
-		}
-		if buildInfo == nil {
-			buildInfo = &model.BuildInfo{
-				VolumesToInclude: svcInfo.VolumeMounts,
-			}
-		} else {
-			buildInfo.Image = svcInfo.Image
+	manifest := model.NewManifestFromStack(s)
+	builder := buildCMD.NewBuildCommand()
+	if options.ForceBuild {
+		buildOptions := &types.BuildOptions{}
+		if err := builder.BuildV2(ctx, manifest, buildOptions); err != nil {
```

Comment:
Would this work for a stack with a v2 manifest?

Reply:
Following up my comment about having `BuildV1` and `BuildV2` in the same command. It would be less confusing from a client perspective to create a single builder and use `builder.Build`